pkg/scheduler/util/pod_resources.go

```diff
@@ -65,11 +63,6 @@ func GetRequestForResource(resource v1.ResourceName, requests *v1.ResourceList,
 		}
 		return requests.Memory().Value()
 	case v1.ResourceEphemeralStorage:
-		// if the local storage capacity isolation feature gate is disabled, pods request 0 disk.
-		if !utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
```

Comment:
the node config isn't propagating here (and couldn't, if the feature could be on in some nodes and off in others)... what is the behavior if the scheduler honors ephemeral storage and nodes do not?

Reply:
(same question in the other places where the scheduler stops checking the feature gate)

Reply:
The pod will fail to schedule since ephemeral-storage allocatable value is not set if feature is disabled (basically it is 0 available).

We put it in the release note that for system cannot support this feature, they cannot set ephemeral-storage request/limit since this feature will not work. I think this is the right behavior, although before when feature is disabled, the request/limit will be dropped silently, but feature is not working.

Reply:
oh, I think I didn't fully address your comment from previous comment. if some nodes set the flag, and some not. If the pod is scheduled to the nodes with the feature disabled, it will fail.

If user does want to support this mixed nodes, we can suggest to add taint to those nodes to avoid pods being scheduled to those nodes.

Reply:
How would it fail on the node? If the feature is disabled in kubelet, shouldn't it ignore the pod's request?.

Reply:
after beta feature gate for LocalStorageCapacityIsolation is removed, the pod request will not be dropped. The scheduler has the logic to check whether request can be satisfied with node allocatable resources. Since node allocatable is 0, it will fail to schedule and in pending state.

Reply:
I think this is a general challenge with many features that require control plane + kubelet. You can't really start using the feature until the nodes have it enabled.

Reply:
But if I understand - the behavior described here is correct - the scheduler will block the pod until a node indicates capacity, at which time it can try to schedule the pod.  Right?

Reply:
It sounds correct to me